buildtest/spider.py

```diff
@@ -20,15 +20,14 @@ def __init__(self, tree=None):
         if tree:
             self.tree = tree
 
-            if not os.getenv("LMOD_DIR"):
-                raise SystemError(
-                    "Cannot find environment variable LMOD_DIR. Please fix your Lmod configuration!"
+            # Lmod can be optionally installed for using modules
+            self.spider_content = []
+            if os.getenv("LMOD_DIR"):
```

Comment:
Should we be explicit to check if path of LMOD_DIR also exist as valid directory. It is likely a user can bypass this and set as environment variable. The spider command is accessible if you load the Lmod module or use the LMOD_DIR to reference the path. By default any site that has Lmod installed as this environment variable setup as part of startup configuration.

Reply:
yes, this is a good idea. I'll add this now. I think I had originally not added so I could try to "trick" buildtest that I had it installed just by exporting the envar. :)